packages/relay/tests/lib/repositories/hbarLimiter/ethAddressPlanRepository.spec.ts

```diff
@@ -0,0 +1,152 @@
+        const ethAddress = '0x123';
+        const addressPlan: IEthAddressHbarSpendingPlan = { ethAddress, planId: uuidV4(randomBytes(16)) };
+        await cacheService.set(`${repository['collectionKey']}:${ethAddress}`, addressPlan, 'test');
+
+        const result = await repository.findByAddress(ethAddress);
+        expect(result).to.deep.equal(addressPlan);
+      });
+
+      it('throws an error if address plan is not found', async () => {
+        const ethAddress = '0xnonexistent';
+        await expect(repository.findByAddress(ethAddress)).to.be.eventually.rejectedWith(
+          EthAddressPlanNotFoundError,
+          `EthAddressPlan with address ${ethAddress} not found`,
+        );
+      });
+    });
+
+    describe('save', () => {
+      it('saves an address plan successfully', async () => {
+        const ethAddress = '0x123';
+        const addressPlan: IEthAddressHbarSpendingPlan = { ethAddress, planId: uuidV4(randomBytes(16)) };
+
+        await repository.save(addressPlan);
+        const result = await cacheService.getAsync<IEthAddressHbarSpendingPlan>(
+          `${repository['collectionKey']}:${ethAddress}`,
+          'test',
+        );
+        expect(result).to.deep.equal(addressPlan);
+      });
+
+      it('overwrites an existing address plan', async () => {
+        const ethAddress = '0x123';
+        const addressPlan: IEthAddressHbarSpendingPlan = { ethAddress, planId: uuidV4(randomBytes(16)) };
+        await cacheService.set(`${repository['collectionKey']}:${ethAddress}`, addressPlan, 'test');
+
+        const newPlanId = uuidV4(randomBytes(16));
+        const newAddressPlan: IEthAddressHbarSpendingPlan = { ethAddress, planId: newPlanId };
+        await repository.save(newAddressPlan);
+        const result = await cacheService.getAsync<IEthAddressHbarSpendingPlan>(
+          `${repository['collectionKey']}:${ethAddress}`,
+          'test',
+        );
+        expect(result).to.deep.equal(newAddressPlan);
+      });
+    });
+
+    describe('delete', () => {
+      it('deletes an address plan successfully', async () => {
+        const ethAddress = '0x123';
+        const addressPlan: IEthAddressHbarSpendingPlan = { ethAddress, planId: uuidV4(randomBytes(16)) };
+        await cacheService.set(`${repository['collectionKey']}:${ethAddress}`, addressPlan, 'test');
+
+        await repository.delete(ethAddress);
+        const result = await cacheService.getAsync<IEthAddressHbarSpendingPlan>(
+          `${repository['collectionKey']}:${ethAddress}`,
+          'test',
+        );
+        expect(result).to.be.null;
+      });
+
+      it('does not throw an error if address plan to delete does not exist', async () => {
+        const ethAddress = '0xnonexistent';
+        await expect(repository.delete(ethAddress)).to.be.fulfilled;
+      });
+    });
+  };
+
+  describe('with shared cache', () => {
+    tests(true);
+  });
+
+  describe('without shared cache', () => {
```

Comment:
Should have a todo note for future refference

Reply:
WDYM? Tests are already implemented here.